### PR TITLE
[libconnman-qt5] Clear default route when it is removed from the cache. 

### PR DESCRIPTION
**Pull request by abranson**
Fixes JB#61664

Route is deleted when connman says it's been removed, which can leave m_defaultRoute as a hanging pointer, which will cause a crash next time the default route changes.

### CODE REVIEW

File: libconnman-qt/networkmanager.cpp [hunk 1]
```diff
@@ -746,13 +746,15 @@ void NetworkManager::updateServices(const ConnmanObjectList &changed, const QLis
...
     for (const QDBusObjectPath &obj : removed) {
         const QString path(obj.path());
-        NetworkService *service = m_servicesCache.value(path);
+        NetworkService *service = m_servicesCache.take(path);
         if (service) {
             if (service == m_priv->m_connectedWifi) {
                 m_priv->m_connectedWifi = NULL;
             }
+            if (m_defaultRoute == service) {
+                m_defaultRoute = NULL;
```
Comment (LaakkonenJussi) [inline libconnman-qt/networkmanager.cpp]: Just wondering that should we just set this invalid like,
```
m_defaultRoute = m_invalidDefaultRoute
```
as that will be the end result.

File: libconnman-qt/networkmanager.cpp [hunk 2]
```diff
@@ -766,8 +768,11 @@ void NetworkManager::updateServices(const ConnmanObjectList &changed, const QLis
         QStringList keys = m_servicesCache.keys();
         for (const QString &path: keys) {
             if (!m_servicesOrder.contains(path)) {
-                m_servicesCache.value(path)->deleteLater();
-                m_servicesCache.remove(path);
+                NetworkService *service = m_servicesCache.take(path);
+                if (m_defaultRoute == service) {
+                    m_defaultRoute = NULL;
```
Comment (LaakkonenJussi) [inline libconnman-qt/networkmanager.cpp]: Same here, -> `m_invalidDefaultRoute` instead of `NULL`?

**Comment by LaakkonenJussi** [comment]
Yeah, this makes sense if some service gets to be the default route and the service gets removed in some cases it may not be updated, like in the user change. A proper corner case that happens just sometimes but seems to be good approach for me, Maybe have [libconnman-qt] here for the commit to avoid confusion?

File: libconnman-qt/networkmanager.cpp (hunk 1 above)
Comment (abranson) [reply libconnman-qt/networkmanager.cpp]: Done. Seems to work fine still. I just noticed that the previous if does a very similar thing, so I'll flip the operands so service is first in this one too.

**Review by LaakkonenJussi — APPROVED** [review]
LGTM.

The PR title still has plain `connman`, though.

@pvuorela any objections here?

**Review by pvuorela — APPROVED** [review]
Looks good to me

**Pull request merged**